heron/uploaders/src/java/com/twitter/heron/uploader/hdfs/HdfsUploader.java

```diff
@@ -65,33 +66,32 @@ protected boolean isLocalFileExists(String file) {
   }
 
   @Override
-  public URI uploadPackage() {
+  public URI uploadPackage() throws UploaderException {
     // first, check if the topology package exists
     if (!isLocalFileExists(topologyPackageLocation)) {
-      LOG.info("Topology file " + topologyPackageLocation + " does not exist.");
-      return null;
+      throw new UploaderException(
+        String.format("Topology file %s does not exist", topologyPackageLocation));
```

Comment:
more context here:

"Expected topology package file to be uploaded %s does not exist"